op-txpool/conditional_txs.go

```diff
@@ -0,0 +1,184 @@
+package op_txpool
+
+import (
+	"context"
+	"fmt"
+	"strings"
+	"time"
+
+	"github.com/ethereum-optimism/optimism/op-service/client"
+	"github.com/ethereum-optimism/optimism/op-service/metrics"
+	"github.com/ethereum-optimism/optimism/op-service/predeploys"
+	oprpc "github.com/ethereum-optimism/optimism/op-service/rpc"
+
+	"github.com/ethereum/go-ethereum/accounts"
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/common/hexutil"
+	"github.com/ethereum/go-ethereum/core/types"
+	"github.com/ethereum/go-ethereum/crypto"
+	"github.com/ethereum/go-ethereum/log"
+	"github.com/ethereum/go-ethereum/rpc"
+
+	"github.com/prometheus/client_golang/prometheus"
+
+	"golang.org/x/time/rate"
+)
+
+var (
+	authHeaderKey = "x-optimism-signature"
```

Comment:
We should be mindful of the existing tooling that just works with the flashbots auth header key, now new tooling must be built to support our small diff